openlibrary/plugins/worksearch/code.py

```diff
@@ -609,17 +610,22 @@ def GET(self):
         return page
 
 
-def works_by_author(akey, sort='editions', page=1, rows=100, has_fulltext=False):
+def works_by_author(akey, sort='editions', page=1, rows=100, has_fulltext=False, query=None):
     # called by merge_author_works
-    q='author_key:' + akey
+    q = 'author_key:' + akey
+    if query:
+        q = query
+
     offset = rows * (page - 1)
     fields = ['key', 'author_name', 'author_key', 'title', 'subtitle',
         'edition_count', 'ia', 'cover_edition_key', 'has_fulltext',
         'first_publish_year', 'public_scan_b', 'lending_edition_s', 'lending_identifier_s',
         'ia_collection_s', 'cover_i']
     fl = ','.join(fields)
-    fq = 'has_fulltext:true' if has_fulltext else ''  # ebooks_only
-    solr_select = solr_select_url + "?fq=type:work&q.op=AND&q=%s&fq=%s&start=%d&rows=%d&fl=%s&wt=json" % (q, fq, offset, rows, fl)
+    fq = ['author_key:OL31676A', 'type:work']
```

Comment:
Looks like we are always adding a specific author key instead we may want to do :
```suggestion
    fq = ['author_key:' + akey, 'type:work']
```

Reply:
great catch